resources/qml/dialogs/RoomSettings.qml

```diff
@@ -29,113 +31,118 @@ ApplicationWindow {
             }
 
-        }
-
-        Connections {
-            target: roomSettings
-            function onDisplayError(errorMessage) {
-                errorText.text = errorMessage;
-                errorText.opacity = 1;
-                hideErrorAnimation.restart();
+            Connections {
+                target: roomSettings
+                function onDisplayError(errorMessage) {
+                    errorText.text = errorMessage;
+                    errorText.opacity = 1;
+                    hideErrorAnimation.restart();
+                }
             }
-        }
-
-        ColumnLayout {
-            Layout.alignment: Qt.AlignHCenter
+                Label {
+                    text: roomSettings.roomName
+                    Layout.alignment: Qt.AlignHCenter
+                    font.pixelSize: fontMetrics.font.pixelSize * 2
+                    Layout.fillWidth: true
+                    horizontalAlignment: TextEdit.AlignHCenter
+                }
 
-            MatrixText {
-                text: roomSettings.roomName
-                font.pixelSize: fontMetrics.font.pixelSize * 2
-                Layout.fillWidth: true
-                horizontalAlignment: TextEdit.AlignHCenter
-            }
+                Label {
+                    text: qsTr("%n member(s)", "", roomSettings.memberCount)
+                    Layout.alignment: Qt.AlignHCenter
 
-            MatrixText {
-                text: qsTr("%n member(s)", "", roomSettings.memberCount)
-                Layout.alignment: Qt.AlignHCenter
+                    TapHandler {
+                        onSingleTapped: TimelineManager.openRoomMembers(Rooms.getRoomById(roomSettings.roomId))
+                    }
 
-                TapHandler {
-                    onSingleTapped: TimelineManager.openRoomMembers(Rooms.getRoomById(roomSettings.roomId))
-                }
+                    CursorShape {
+                        cursorShape: Qt.PointingHandCursor
+                        anchors.fill: parent
+                    }
 
-                CursorShape {
-                    cursorShape: Qt.PointingHandCursor
-                    anchors.fill: parent
                 }
 
+            ImageButton {
+                Layout.alignment: Qt.AlignHCenter
+                image: ":/icons/icons/ui/edit.svg"
+                visible: roomSettings.canChangeNameAndTopic
+                onClicked: roomSettings.openEditModal()
             }
 
-        }
-
-        ImageButton {
-            Layout.alignment: Qt.AlignHCenter
-            image: ":/icons/icons/ui/edit.svg"
-            visible: roomSettings.canChangeNameAndTopic
-            onClicked: roomSettings.openEditModal()
-        }
-
-        ScrollView {
-            Layout.fillHeight: true
-            Layout.alignment: Qt.AlignHCenter
-            Layout.fillWidth: true
-            Layout.leftMargin: Nheko.paddingLarge
-            Layout.rightMargin: Nheko.paddingLarge
-
             TextArea {
```

Comment:
How does this scroll now? I have a few topics that are over 100 lines...

Reply:
You can check the first commit, those are all the relevant changes.
It scrolls the entire contents with the exception of the OK button, just like UserProfile. Is that what you meant?

Reply:
Well, that makes it very annoying, when there is a large topic. Either make that collapse/uncollapse or put it in something scrollable.

Reply:
mh, I never had a large topic I guess. I suppose collapse/uncollapse is the way to go, because you don't want scrollable on top of scrollable. I'll have to think of a good way to do it though.
That would make the topic much nicer to read, if you actually wanted to ;)